docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -287,19 +308,190 @@ void TemplateInferRequest::waitPipeline() {
 void TemplateInferRequest::inferPostprocess() {
     OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, _profilingTask[Postprocess]);
     auto start = Time::now();
-    for (auto&& output : _outputs) {
-        auto outputBlob = output.second;
+    for (auto&& output : _networkOutputs) {
+        auto index = _executableNetwork->_outputIndex[output.first];
+        const auto& result = _executableNetwork->_function->get_results()[index];
+        if (result->get_output_partial_shape(0).is_dynamic()) {
+            // Touch blob to allocate it
+            Blob::Ptr blob;
+            GetBlob(output.first);
+        }
+        auto outputBlob = _outputs.at(output.first);
         auto networkOutput = _networkOutputBlobs[output.first];
-        // perform precision conversion of network output's precision and computational
-        // graph output's precision are different
         if (outputBlob->getTensorDesc().getPrecision() != networkOutput->getTensorDesc().getPrecision()) {
             blobCopy(networkOutput, outputBlob);
+        } else if (result->get_output_partial_shape(0).is_dynamic()) {
+            auto tensor = _outputTensors[_executableNetwork->_outputIndex.at(output.first)];
+            tensor->read(InferenceEngine::as<InferenceEngine::MemoryBlob>(outputBlob)->wmap().as<char*>(),
+                         tensor->get_size_in_bytes());
         }
     }
     _durations[Postprocess] = Time::now() - start;
 }
 // ! [infer_request:infer_postprocess]
 
+// ! [infer_request:get_blob]
+InferenceEngine::Blob::Ptr TemplateInferRequest::GetBlob(const std::string& name) {
```

Comment:
we also need to update documentation https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_PLUGIN_DG/InferRequest.md